src/uu/mktemp/src/mktemp.rs

```diff
@@ -191,7 +191,22 @@ impl Options {
                     (tmpdir, template.to_string())
                 }
                 Some(template) => {
-                    let tmpdir = matches.get_one::<String>(OPT_TMPDIR).map(String::from);
+                    let tmpdir = if let Ok(tmpdir_env) = env::var("TMPDIR") {
```

Comment:
I think you need an additional condition here. It currently breaks the GNU tests because GNU mktemp ignores `TMPDIR` in the following case:
```
$ TMPDIR=dir_a mktemp foo.XXXX
foo.fgPS
```